### PR TITLE
[Fixes #546] respects it’s host apps source-map config.

### LINKED ISSUE #546: respect disabled sourcemaps from host
setting `sourcemaps.enabled = false` in dev mode from host will still create sourcemaps in the engine.

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by rwjblue** [comment]
Seems odd to hard code this to false (vs reading the value from the top level ember-cli-build.js). Doesn’t seem like this PR does what the title suggests (reading the config from the host app), did I just miss some nuance?

**Comment by stefanpenner** [comment]
investing:

```js
not ok 6 Acceptance build correctly builds lazy engine in lazy engine
  AssertionError: expected { Object (bundles) } to deeply equal { Object (bundles) }
      at Context.<anonymous> (node-tests/acceptance/build-test.js:262:43)
```

**Comment by villander** [comment]
we will need add tests on this scenario @stefanpenner

File: lib/engine-addon.js [hunk 1]
```diff
@@ -604,6 +608,9 @@ module.exports = {
         let vendorCSSTree = buildVendorCSSTree.call(this, vendorTree);
         let engineStylesOutputDir = 'engines-dist/' + this.name + '/assets/';
 
+        let hostOptions = findHostsHost.call(this).options || {};
+        let sourceMapConfig = hostOptions.sourcemaps || { enabled: false };
```
[the remark this answers is not in the document]
Comment (stefanpenner) [reply lib/engine-addon.js]: @villander can you elaborate, I want to make sure I address your concern correctly.

File: node-tests/acceptance/build-test.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ describe('Acceptance', function() {
         });
       });
 
-    it(
+    it.only(
```
Comment (rwjblue) [inline node-tests/acceptance/build-test.js]: plz to 🔨

File: node-tests/acceptance/build-test.js [hunk 3]
```diff
@@ -66,6 +66,7 @@ describe('Acceptance', function() {
 
         let output = await build(app);
 
+        debugger;
```
Comment (rwjblue) [inline node-tests/acceptance/build-test.js]: ✂️

Comment (stefanpenner) [reply node-tests/acceptance/build-test.js]: ya, currently fixing this test scenario. I'll remove, as it didn't need to actually be committed.

File: lib/engine-addon.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (villander) [reply lib/engine-addon.js]: I think it's all correct now, as I was missing tests I was scared of this fallback breaking the build

thank you for this work @stefanpenner

File: node-tests/fixtures/expected-manifests.json [hunk 4]
```diff
@@ -11,10 +11,6 @@
     "bundles": {
       "lazy": {
         "assets": [
-          {
```
Comment (villander) [inline node-tests/fixtures/expected-manifests.json]: @stefanpenner this lines removed no longer exist on ember engines build?

File: lib/engine-addon.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (stefanpenner) [reply lib/engine-addon.js]: I finally have some cycles to land this. I'll try and wrap this one up today.

File: node-tests/fixtures/expected-manifests.json (hunk 4 above)
Comment (stefanpenner) [reply node-tests/fixtures/expected-manifests.json]: this was just local testing that I committed. We are good

**Comment by villander** [comment]
thank you @stefanpenner I'll to document this on `ember-engines.com`

**Comment by stefanpenner** [comment]
@villander i think the ember-cli docs should cover this (or maybe they need to be updated). This should simply ensure ember-engines was a good citizen.

**Comment by villander** [comment]
Yes @stefanpenner, [ember-cli docs](https://cli.emberjs.com/release/) does not say anything about ember engines, unfortunately there is a big gap between ember-engines with ember guides

**Comment by stefanpenner** [comment]
tested locally in our mega app, all appears well.

**Comment by stefanpenner** [comment]
+ ember-engines@0.6.2

**Pull request merged**